rust/src/tests/errors.rs

```diff
@@ -0,0 +1,589 @@
+use nix::{
+    errno::Errno,
+    fcntl::{open, OFlag},
+    libc::off_t,
+    sys::stat::{mknod, Mode, SFlag},
+    unistd::{chown, ftruncate, mkdir, mkfifo, truncate, unlink, Uid, User},
+};
+
+#[cfg(any(
+    target_os = "openbsd",
+    target_os = "netbsd",
+    target_os = "freebsd",
+    target_os = "dragonfly",
+    target_os = "macos",
+    target_os = "ios"
+))]
+use nix::{sys::stat::FileFlag, unistd::chflags};
+
+use std::{
+    fmt::Debug,
+    panic::AssertUnwindSafe,
+    path::{Path, PathBuf},
+};
+
+use crate::{
+    runner::context::{FileType, SerializedTestContext, TestContext},
+    tests::{chmod, lchmod, lchown, link, rename, symlink},
+};
+
+crate::test_case! {eloop}
+fn eloop(ctx: &mut TestContext) {
```

Comment:
This test combines way too many assertions into a single test case.  Could you split it up?